File: sdks/php/src/AddOns/Requests/UpdateAddOnDto.php
<?php

namespace NovaBilling\AddOns\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Types\AddOnPriceDto;
use NovaBilling\Core\Types\ArrayType;

class UpdateAddOnDto extends JsonSerializableType
{
    /**
     * @var ?string $name
     */
    #[JsonProperty('name')]
    public ?string $name;

    /**
     * @var ?string $description
     */
    #[JsonProperty('description')]
    public ?string $description;

    /**
     * @var ?string $invoiceDisplayName
     */
    #[JsonProperty('invoiceDisplayName')]
    public ?string $invoiceDisplayName;

    /**
     * @var ?array<AddOnPriceDto> $prices
     */
    #[JsonProperty('prices'), ArrayType([AddOnPriceDto::class])]
    public ?array $prices;

    /**
     * @param array{
     *   name?: ?string,
     *   description?: ?string,
     *   invoiceDisplayName?: ?string,
     *   prices?: ?array<AddOnPriceDto>,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->name = $values['name'] ?? null;
        $this->description = $values['description'] ?? null;
        $this->invoiceDisplayName = $values['invoiceDisplayName'] ?? null;
        $this->prices = $values['prices'] ?? null;
    }
}

File: sdks/ruby/lib/novabilling/add_ons/types/update_add_on_dto.rb
# frozen_string_literal: true

module Novabilling
  module AddOns
    module Types
      class UpdateAddOnDto < Internal::Types::Model
        field :id, -> { String }, optional: false, nullable: false
        field :name, -> { String }, optional: true, nullable: false
        field :description, -> { String }, optional: true, nullable: false
        field :invoice_display_name, -> { String }, optional: true, nullable: false, api_name: "invoiceDisplayName"
        field :prices, -> { Internal::Types::Array[Novabilling::Types::AddOnPriceDto] }, optional: true, nullable: false
      end
    end
  end
end
